src/main/java/com/metamx/common/io/smoosh/FileSmoosher.java

```diff
@@ -65,9 +68,12 @@
 
   private final List<File> outFiles = Lists.newArrayList();
   private final Map<String, Metadata> internalFiles = Maps.newTreeMap();
-
+  private Map<String, File> createdFiles = Maps.newLinkedHashMap();
+  private ArrayDeque<File> files = Queues.newArrayDeque();
```

Comment:
I'm not sure the fact that this is a `Deque` is so important, can we just make it a `List` and then after iterating through it just call `clear()` on it?